src/Microsoft.DotNet.GitHub.Authentication/GitHubTokenProvider.cs

```diff
@@ -84,9 +84,9 @@ private bool TryGetCachedToken(long installationId, out AccessToken cachedToken)
 
         AccessToken token = _tokenCache[installationId];
 
-        // If the cached token will expire in less than 30 minutes we won't use it and let GetTokenForInstallationAsync generate a new one
+        // If the cached token will expire in less than 50 minutes we won't use it and let GetTokenForInstallationAsync generate a new one
         // and update the cache
-        if (token.ExpiresAt.Subtract(DateTimeOffset.Now).TotalMinutes < 30)
+        if (token.ExpiresAt.Subtract(DateTimeOffset.Now).TotalMinutes < 50)
```

Comment:
We looked  into it and we ended up on it not having the UtcNow

```suggestion
        if (token.ExpiresAt.Subtract(DateTimeOffset.UtcNow).TotalMinutes < 15)
```